neo.UnitTests/UT_UIntBenchmarks.cs

```diff
@@ -0,0 +1,140 @@
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Ledger;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Diagnostics;
+using System;
+//using System.Runtime.CompilerServices.Unsafe;
+
+namespace Neo.UnitTests
+{
+    [TestClass]
+    public class UT_UIntBenchmarks
+    {
+        int MAX_TESTS = 1000000; // 1 million
+
+        byte[][] base_32_1;
+        byte[][] base_32_2;
+        byte[][] base_20_1;
+        byte[][] base_20_2;
+
+        private Random random;
+
+        [TestInitialize]
+        public void TestSetup()
+        {
+            int SEED = 123456789;
+            random = new Random(SEED);
+
+            base_32_1 = new byte[MAX_TESTS][];
+            base_32_2 = new byte[MAX_TESTS][];
+            base_20_1 = new byte[MAX_TESTS][];
+            base_20_2 = new byte[MAX_TESTS][];
+
+            for(var i=0; i<MAX_TESTS; i++)
+            {
+                base_32_1[i] = RandomBytes(32);
+                base_32_2[i] = RandomBytes(32);
+                base_20_1[i] = RandomBytes(20);
+                base_20_2[i] = RandomBytes(20);
+            }
+        }
+
+        private byte[] RandomBytes(int count)
+        {
+            byte[] randomBytes = new byte[count];
+            random.NextBytes(randomBytes);
+            return randomBytes;
+        }
+
+        [TestMethod]
+        public void Benchmark_CompareTo()
+        {
+            // testing "official version"
+            UInt256[] uut_32_1 = new UInt256[MAX_TESTS];
+            UInt256[] uut_32_2 = new UInt256[MAX_TESTS];
+
+            for(var i=0; i<MAX_TESTS; i++)
+            {
+                uut_32_1[i] = new UInt256(base_32_1[i]);
+                uut_32_2[i] = new UInt256(base_32_2[i]);
+            }
+            Stopwatch sw0 = new Stopwatch();
```

Comment:
Can you make a method that runs the stopwatch and takes a delegate with code to be benchmarked. The loop would occur in the passed delegate.

Reply:
Since times are quite short... we need to measure them all together, if I measure one by one, system time will be bigger than timing itself.

Reply:
No I wasn’t suggesting measure 1 by 1. I was suggesting a benchmark method that takes a delegate. The passed delegate has the code that loops 1 million times all in one call to the delegate.

Reply:
oh!! nice! it would look better indeed for better isolation.